design/synchronizer/auto-failover.md

```diff
@@ -0,0 +1,92 @@
+# Synchronizer Auto-Failover
+
+The following is an overview of the workflow for a Vault Sychronizer to gain leadership:
+
+![](./activity-diagram.png)
+
+Given the following Policy:
+```yml
+- !policy
+  id: conjur
+  body:
+  - !policy
+    id: vault-synchronizer
+    body:
+    - !variable leader-expiration
+
+    # The Group all Synchronizer hosts need to be added to.
+    - !group members
+
+    - !permit
+        role: !group members
+        privileges: [ read, execute, update ]
+        resources:
+      - !variable leader-expiration
+
+      - !policy
+        id: leader
+        owner: !group members
+
+      - !policy
+        id: vault-lines-of-business
+        owner: !group members
+        body:
+        - !variable
+              id: lob-1
+              mime_type: application/json
+        - !variable
+              id: lob-2
+              mime_type: application/json
+```
+
+The following are the steps (using the API) listed above:
+
+1. Update `leader-expiration` to future time:
+
+    ```sh
+    POST /secrets/{account}/variable/conjur%2Fvault-synchronizer%2Fleader-expiration
+
+    Time.now() + <time in seconds> // Save as Unix epoch time
+    ```
+
+1. `leader-expiration` time has passed:
+
+    ```sh
+    GET /secrets/{account}/variable/conjur%2Fvault-synchronizer%2Fleader-expiration
+    ```
+
+    Then compare returned Unix epoch time time to current Unix epoch time.
+
+1. Delete leader host:
```

Comment:
What happens in the very first flow:
- no leader
- trying to delete vault-synchronizer-leader if that doesn't exist?

Should I check if that variable is set altogether with expiration date or can I just try to delete it and then try to create it.

Reply:
Delete on a resource that's not present results in a no-op, so it should be fine to always attempt the delete before trying to claim "leader" status.